sdk/identity/Azure.Identity/src/ImdsManagedIdentityProbeSource.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Identity
+{
+    internal class ImdsManagedIdentityProbeSource : ManagedIdentitySource
+    {
+        // IMDS constants. Docs for IMDS are available at https://learn.microsoft.com/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http
```

Comment:
Likely orthogonal to the intended scope of the change, but I wonder if it would be helpful to link out to some doc that describes the common MI issues.   We're seeing a lot of "credential says no MI endpoint but when I SSH and curl it, the endpoint works for me."

It would be amazing to have a place to point people for "go talk to the owners of your host environment."

Reply:
We genrally do that via our [troubleshooting doc](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/TROUBLESHOOTING.md), which gets linked in most exceptions. As we find new common scenarios, we add them there.